zebedee-cms/src/main/java/com/github/onsdigital/zebedee/util/slack/StartUpAlerterImpl.java

```diff
@@ -40,18 +50,46 @@ public StartUpAlerterImpl(SlackClient slack, List<String> channels) {
     StartUpAlerterImpl(SlackClient slack, List<String> channels, boolean queueLocked,
                        boolean lockedNotificationSent) {
         this.slack = slack;
-        this.channels = channels;
         this.queueLocked = queueLocked;
         this.lockedNotificationSent = lockedNotificationSent;
-        this.messages = new ArrayList<>();
+
+        this.queueLockedAlerts = channels.stream()
+                .map(c -> newQueueLockedAlertTask(c))
+                .collect(Collectors.toList());
+
+        this.queueUnlockedAlerts = new ArrayList<>();
+
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            info().log("shutting down slack thread executor service");
+            executorService.shutdown();
+        }));
     }
 
     @Override
     public void queueLocked() {
         synchronized (MUTEX) {
             if (queueLocked && !lockedNotificationSent) {
-                channels.stream().forEach(c -> messages.add(publishingQueueLocked(c)));
+
+                List<Future<PostMessage>> futures = null;
+                try {
+                    futures = executorService.invokeAll(queueLockedAlerts);
+                } catch (Exception ex) {
+                    error().exception(ex).log("error sending publishing queue locked slack notification");
```

Comment:
Being pedantic, but the error is in creating the futures that will send notifications, not with sending the notifications themselves.